### PR TITLE
New Data Source `azurerm_managed_api` & New Resource: `azurerm_api_connection`

### LINKED ISSUE #1691: Logic Apps: support for connections
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add into Logic App Resource a resource that handles api connection creation and edition.

### New or Affected Resource(s)

azurerm_logic_app_api_connection 
(suggestion)

### PR DESCRIPTION
**Pull request by tombuildsstuff**
This PR introduces support for creating API Connections, which also required creating a Data Source to retrieve the Managed API information.

Whilst this creates the API Connection resource this doesn't handle linking these to other resources (such as Logic Apps) - which'll need to be tracked in a separate issue, but allowing the creation of these is a step forward at any rate.

Unfortunately the API Version used in the Azure Portal doesn't have associated Swagger in the Azure Rest API Specs repository, as such I've had to use an older API version which doesn't allow the update of Parameters - which (whilst obviously not ideal) seemed like a worthwhile trade-off here.

Fixes #1691

### CODE REVIEW

File: website/docs/r/api_connection.html.markdown [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  tags = {
+    Hello = "World"
+  }
+
+  lifecycle {
+    # NOTE: since the connectionString is a secure value it's not returned from the API
+    ignore_changes = ["parameter_values"]
+  }
+}
+```
```
Comment (manicminer) [inline website/docs/r/api_connection.html.markdown:35]: Looks like the properties/attributes/import sections are missing?

**Review by manicminer — COMMENTED** [review]
One question about the docs for `azurerm_api_connection` but otherwise LGTM 👍

File: website/docs/r/api_connection.html.markdown (hunk 1 above)
Comment (tombuildsstuff) [reply website/docs/r/api_connection.html.markdown:35]: 🤦 my bad / good spot - thanks

**Comment by tombuildsstuff** [comment]
Tests pass:

Screenshot 2022-03-14 at 15 16 35

**Pull request merged**